### PR TITLE
[Search] clean up search-react api report warnings

### PR DESCRIPTION
**Pull request by emmaindal**
Signed-off-by: Emma Indal <emma.indahl@gmail.com>

## Hey, I just made a Pull Request!

This PR cleans up all warnings in the api-report of the search-react package.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by iamEAP — APPROVED** [review]
✨ ✨ ✨

File: plugins/search-react/src/context/SearchContextForStorybook.stories.tsx [hunk 1]
```diff
@@ -18,35 +18,54 @@ import { SearchResultSet } from '@backstage/plugin-search-common';
...
-export const SearchContextProvider = (
-  props: ComponentProps<typeof RealSearchContextProvider> & QueryResultProps,
-) => {
-  return (
-    <SearchApiProvider {...props}>
-      <RealSearchContextProvider children={props.children} />
-    </SearchApiProvider>
-  );
-};
+export type SearchContextProviderForStorybookProps = PropsWithChildren<{
```
Comment (Rugvip) [inline plugins/search-react/src/context/SearchContextForStorybook.stories.tsx:37]: We'd typically export a mock API implementation instead, which you can use both in tests, dev environments, or storybook. You'd use the `TestApiProvider` to inject it in the storybook.

Already here though so not a blocker for the PR, just ideas for the future

**Review by Rugvip — APPROVED** [review]
🧹 ✨

**Comment by iamEAP** [comment]
Gonna merge this in, but noted!

**Pull request merged**